internal/plans/objchange/objchange.go

```diff
@@ -345,8 +347,10 @@ func proposedNewNestedType(schema *configschema.Object, prior, config cty.Value)
 			} else {
 				newV = cty.ListVal(newVals)
 			}
-		} else {
-			newV = cty.NullVal(schema.ImpliedType())
+		case config.IsNull():
+			newV = cty.NullVal(config.Type())
+		default:
+			newV = cty.ListValEmpty(config.Type().ElementType())
```

Comment:
It looks like these two cases could be safely simplified down to just one like this:

```go
default:
    newV = config
```

...since right now both of those paths seem to just be constructing a new value equivalent to the one we're already holding, right?

(Similar question for the other variants of this below.)

Reply:
🤦‍♂️Thank you, that is a correct simplification. I was repeating a pattern used elsewhere when creating collections based on other constraints, but it’s unnecessary here since we have a value already.